src/puppetlabs/trapperkeeper/services/authorization/authorization_service.clj

```diff
@@ -20,6 +21,9 @@
           (assoc-in [:allow-header-cert-info] (get config
                                                    :allow-header-cert-info
                                                    false)))))
+  (authorization-check [this request]
+   (let [{:keys [rules allow-header-cert-info]} (service-context this)]
+    (ring-middleware/authorization-check request rules allow-header-cert-info)))
```

Comment:
Yes, the core functions should not have to adjust or modify the request in any way.  And I agree we need to account for the query-params with this new function.  :+1: for having something in the HTTP/ring level do this before calling into core.